code/backend/batch/utilities/helpers/AzureBlobStorageHelper.py

```diff
@@ -131,9 +121,10 @@ def get_all_files(self):
         return files
 
     def upsert_blob_metadata(self, file_name, metadata):
-        blob_client = BlobServiceClient.from_connection_string(
-            self.connect_str
-        ).get_blob_client(container=self.container_name, blob=file_name)
+        if self.auth_type == 'rbac':
+            blob_client = self.blob_service_client.get_blob_client(container=self.container_name, blob=file_name)
```

Comment:
Just curious, why not make the blob service client in the constructor? For timing reasons?

Reply:
`blob_service_client` is already in the constructor, what do you want to say is `blob_client`? This is the author's initial design, and we have not changed it.

Reply:
Oh sorry thats what I meant. Okay, that's fine, was just curious.